### PR TITLE
Support reload certs in HTTPS server.

### PR DESCRIPTION
**Pull request by yswdqz**
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [ ] Update the documentation to include this new feature.
- [ ] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

**Comment by yswdqz** [comment]
I forget add a unit test, I will do it later.

**Comment by wu-sheng** [comment]
I think usually we need a manual test. I don't think tls could be easily tested. 
Wait for @kezhenxu94

**Comment by kezhenxu94** [comment]
> I think usually we need a manual test. I don't think tls could be easily tested. 
> Wait for @kezhenxu94 

To test the reloading mechanism, we can recreate new cert pairs with scripts and update them in test and verify the API still works before and after the update, but I'm totally ok if you don't want to test this case. But we need to add a basic case that verify the https works, the same situation exists in our gRPC agent protocol server in this repo. You can take that as an example.

File: oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServer.java [hunk 1]
```diff
@@ -107,15 +149,51 @@ public void addHandler(Object handler, List<HttpMethod> httpMethods) {
...
+            try (InputStream cert = new FileInputStream(tlsCertChainPath.toFile());
+                 InputStream key = PrivateKeyUtil.loadDecryptionKey(tlsKeyPath.toString())) {
+                sb.tls(cert, key);
+            } catch (IOException e) {
+                throw new IllegalArgumentException(e);
+            }
+            normalInitialize(sb);
+            sb.annotatedService()
+              .pathPrefix(config.getContextPath())
+              .build(handlers.toArray());
```
Comment (kezhenxu94) [inline oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServer.java:185]: Can you confirm this is needed when we only want to reconfigure the tls cert? I think Armeria will maintain the existing service endpoint if we didn't add new or remove existing service.

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]
According to https://github.com/line/armeria/pull/4637 this PR can't be merged until they release a new version containing the mentioned Armeria PR.

File: oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServer.java (hunk 1 above)
Comment (yswdqz) [reply oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServer.java:185]: `reconfigure` method will creat a new ServerBuild. So I think it is needed.
image
image

Comment (yswdqz) [reply oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServer.java:185]: `sb.buildServerConfig(config())` only set the host and port.

[unit test of Armeria](https://github.com/line/armeria/blob/871d87297e4d051241589cb1ae95641cbc83f880/core/src/test/java/com/linecorp/armeria/client/ReconfigurableServerTest.java#L120) can also prove it.

**Comment by yswdqz** [comment]
I have not tested it now. I will test it after they `armeria` release a new version.

**Comment by kezhenxu94** [comment]
Armeria 1.22 is released you can continue this. https://armeria.dev/release-notes/1.22.0/

**Comment by yswdqz** [comment]
Got it , But when I was writing tests for dynamodb, I found that after loading the certificate on skywalking, aws would not successfully send data to the endpoint, while when loading the certificate on nginx and routing it to skywalking, aws could successfully send data to the endpoint. So the existing https server may have some bugs, I will test and finish this PR after the PR about dynamodb is finished

**Comment by wu-sheng** [comment]
By following https://github.com/apache/skywalking/pull/10409, we don't need to support TLS.
Close this too.

**Pull request closed without merging**